test/serializer/additional-functions/MutatedObject.js

```diff
@@ -0,0 +1,15 @@
+function additional1(x, y) {
```

Comment:
So this test doesn't reach a fixed point. Looking at the code generated for the different iterations, it looks remarkably similar! Except... The following flip-flops.
```js
var _E = _$2 === _$3;
```
```js
var _E = _$3 === _$2;
```

Reply:
Not sure exactly how it happens. Does the CSE have some such flip-flopping build into the algorithm? @hermanventer?

Reply:
The simplifier does try to make things canonical and that can involve re-ordering commutative operands.

Reply:
Then should we change the tests to not error on the re-ordering of operands?

Reply:
I don't think we should accept non convergence of the fixed point loop. But I'll need to dig into this a bit more.

Reply:
The non convergence is a result of a hash depending on a name that is generated differently (by design) for every fixed point iteration. Fixing the hash to use a simple counter make the failure go away.

To fix this, change line 952 in utils/generator.js from
  let res = new Constructor(this.realm, types, values, hashString(id.name), [], id, options);
to
  let res = new Constructor(this.realm, types, values, this.preludeGenerator.derivedIds.size, [], id, options);

Reply:
Thanks @hermanventer, I've made that change.